packages/stripe-integration/src/stripe-upe/stripe-ocs-payment-strategy.ts

```diff
@@ -0,0 +1,418 @@
+                    fontSize: titleFontSize,
+                    fontWeight: titleFontWeight,
+                    padding: '13px 20px 13px 18px',
+                },
+                '.TabLabel, .AccordionItem': {
+                    fontSize: titleFontSize,
+                    fontWeight: titleFontWeight,
+                    color: titleColor,
+                },
+                '.TabLabel--selected, .AccordionItem--selected': {
+                    fontWeight: titleFontWeight,
+                    color: titleColor,
+                },
+                '.RadioIcon': {
+                    // 26px / 0.88 = 29.54
+                    width: '29.54px',
+                },
+                '.RadioIconOuter': {
+                    // 1px / 26px * 0.88 = 0.034
+                    strokeWidth: '3.4',
+                    stroke: radioColor,
+                },
+                '.RadioIconOuter--checked': {
+                    stroke: radioFocusColor,
+                },
+                '.RadioIconInner': {
+                    // 17.16px / 26px * 0.88 = 0.58 / 2 = 29
+                    r: '29',
+                    fill: radioFocusColor,
+                },
+            },
+        };
+    }
+
+    private _collapseStripeElement() {
+        const stripeElement = this.stripeElements?.getElement(StripeElementType.PAYMENT);
+
+        stripeElement?.collapse();
+    }
+
+    private _getPaymentPayload(methodId: string, token: string): Payment {
+        const cartId = this.paymentIntegrationService.getState().getCart()?.id || '';
+        const formattedPayload = {
+            cart_id: cartId,
+            credit_card_token: { token },
+            confirm: false,
+            payment_method_id: this.selectedMethodId,
+        };
+
+        return {
+            methodId,
+            paymentData: {
+                formattedPayload,
+            },
+        };
+    }
+
+    private async _processAdditionalAction(
+        error: unknown,
+        methodId: string,
+    ): Promise<PaymentIntegrationSelectors | never> {
+        if (
+            !isRequestError(error) ||
+            !this.stripeUPEIntegrationService.isAdditionalActionError(error.body.errors)
+        ) {
+            throw error;
+        }
+
+        if (!this.stripeUPEClient || !this.stripeElements) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        const {
+            data: { token, redirect_url },
+        } = error.body.additional_action_required;
+        const stripePaymentData = this.stripeUPEIntegrationService.mapStripePaymentData(
+            this.stripeElements,
+            redirect_url,
+        );
+        const isPaymentCompleted = await this.stripeUPEIntegrationService.isPaymentCompleted(
+            methodId,
+            this.stripeUPEClient,
+        );
+
+        if (
+            this.stripeUPEIntegrationService.isRedirectAction(error.body.additional_action_required)
```

Comment:
In which cases we will have `isRedirectAction` and `isOnPageAdditionalAction`?

Reply:
In response of first payment request we should receive information witch  kind of confirmation we will have.

For redirect confirmation we will have response like:
```
{
  body: {
    additional_action_required: {
      type: 'redirect_to_url',
      data: {
        redirect_url: '...some url...',
      }
    }
  }
}
```
and in this case we will check `type` and if `redirect_url` exists [isRedirectAction](https://github.com/bigcommerce/checkout-sdk-js/pull/2710/files#diff-c769960fe5cfbe07fa332c83ac27f1857b6d1c2f5bbe8e0d19ad45763998233aR188)

other vice if it is confirmation in modal window, response will be like
```
{
  body: {
    additional_action_required: {
      type: 'additional_action_requires_payment_method',
      data: {
        token: '...some token...',
      }
    }
  }
}
```
and we will check also the `type` and if `token` exists - [isOnPageAdditionalAction](https://github.com/bigcommerce/checkout-sdk-js/pull/2710/files#diff-c769960fe5cfbe07fa332c83ac27f1857b6d1c2f5bbe8e0d19ad45763998233aR197)